### ISSUE TITLE
Problems with make and libtool after trying to add new module

### ISSUE THREAD

**Issue by alexander-mead** [LSSTDESC/CCL#383]
I am having some trouble compiling the new branch I have created. The error I am getting on running `make` is:

`
ld: library not found for -lccl
collect2: error: ld returned 1 exit status
make[2]: *** [tests/check_ccl] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
`

I think that for some reason the CCL library is not being created at the first stage of the `make` command. If I compare the output of `make` run on the master branch against that run on my branch the version on the master branch runs a whole load of `libtool` commands before running the `C` compile commands, while on my branch only the `C` compile commands are being run. I think it is these `libtool` commands that first make the CCL library... 

Can anybody help me out with this? What could I have broken to stop `libtool` being run and to stop the library being made?

**Extra info**

If I `diff` the two version of `Makefile.am` they only differ by the extra module I have added.

If I `diff` the two version of `Makefile.in` then there are quite a lot of differences, I think because I used `automake` 1.16 to regenerate this file (following CONTRIBUTING.md) and also because the new module is in there.

If I `diff` the two versions of `configure` there are quite a lot of differences, I think because I re-ran `autoreconf -i`.

If I `diff` the two versions of `configure.ac` there are no differences.

**Comment by alexander-mead** [comment]
UPDATE

This can be fixed by @philbull doing the following:

```
make clean
autoreconf -i
./configure
make
make install
make check
```

but if I repeat these steps on my Mac the problem persists. However, if I pull a branch that Phil has run these steps on it works fine for me and I can continue my development. 

I think we are both running the same version of `autoreconf` (v 2.69).

**Comment by alexander-mead** [comment]
This problem was solved by me downgrading my version of `automake` from 1.16.1 to 1.15.1. If I do the downgrade and then re-run `autoreconf -i` after doing `make clean` then everything compiles and installs correctly.

Note that this problem was not to do with my branch. It was caused (somehow) by the newer version of `autoreconf`. I was able to reproduce the problem in a clean clone of the master branch.
